### PR TITLE
Update to Bump Hunter Code

### PR DESCRIPTION
**Pull request by mccaky**
A quick update to the bump hunter code. Changes include:

- The mass resolution may now be scaled by a decimal factor from the command line.
- A histogram file and histogram name may be provided as a signal injection source. Monte Carlo will be generated according to the histogram distribution instead of a Gaußian.
- The χ^2 values for the signal+background and toy generator fits are now persisted in the output ntuple.

### CODE REVIEW

File: analysis/include/BumpHunter.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ class BumpHunter {
         };
         
         /** Default Constructor */
-        BumpHunter(BkgModel model, int poly_order, int res_factor, bool asymptotic_limit = true);
+        BumpHunter(BkgModel model, int poly_order, int res_factor, double res_scale = 1.56, bool asymptotic_limit = true);
```
Comment (cbravo135) [inline analysis/include/BumpHunter.h]: Seems dangerous to me for this to default to anything except 1.

File: processors/include/BhToysHistoProcessor.h [hunk 2]
```diff
@@ -77,6 +88,9 @@ class BhToysHistoProcessor : public Processor {
         // of the invariant mass distribution. The number of events may be modified
         // by a multiplicative factor.
         int bkg_mult_{1};
+
+        // The factor by which to scale the mass resolution function.
+        double res_scale_{1.56};
```
Comment (cbravo135) [inline processors/include/BhToysHistoProcessor.h]: This should probably be 1.00 to be safe.

**Review by cbravo135 — CHANGES_REQUESTED** [review]
The default resolution scale should be 1.00

File: analysis/include/BumpHunter.h (hunk 1 above)
Comment (mccaky) [reply analysis/include/BumpHunter.h]: 1.56 was the correct default for Sebouh's resolution, but is not correct for Rafo's. I had updated it elsewhere but forgot to do it here. It should be fixed now.

File: processors/include/BhToysHistoProcessor.h (hunk 2 above)
Comment (mccaky) [reply processors/include/BhToysHistoProcessor.h]: This default was all over the place... Fixed it here, too.

**Review by cbravo135 — APPROVED** [review]
Thanks Kyle!

**Pull request merged**